rustls/src/record_layer.rs

```diff
@@ -181,16 +180,19 @@ impl RecordLayer {
         self.trial_decryption_len = None;
     }
 
-    /// Return true if we are getting close to encrypting too many
-    /// messages with our encryption key.
-    pub(crate) fn wants_close_before_encrypt(&self) -> bool {
-        self.write_seq == SEQ_SOFT_LIMIT
-    }
-
-    /// Return true if we outright refuse to do anything with the
-    /// encryption key.
-    pub(crate) fn encrypt_exhausted(&self) -> bool {
-        self.write_seq >= SEQ_HARD_LIMIT
+    /// Return a remedial action when we are near to encrypting too many messages.
+    ///
+    /// `add` is added to the current sequence number.  `add` as `0` means
+    /// "the next message processed by `encrypt_outgoing`"
+    pub(crate) fn pre_encrypt_action(&self, add: u64) -> PreEncryptAction {
```

Comment:
Would this be easier to understand at call-sites if there was a separate `pub(crate) fn next_pre_encrypt_action()` that calls `pre_encrypt_action(0)` (or similar)?